examples/platformer/code/gamestateplay.pas

```diff
@@ -1375,9 +1387,10 @@ procedure TStatePlay.Update(const SecondsPassed: Single; var HandleInput: Boolea
   inherited;
   { This virtual method is executed every frame.}
 
-  if GameOver then
+  { If player is dead and we did not show game over state we do that }
```

Comment:
I do not see the point of using `Push` here. It makes code harder to reason about: underneath `StateGameOver` you will have, still playing, `StatePlay`, and you need to account for it.

Let's keep the demo simple, and just have `TUIState.Current := StateGameOver`.

And remove the extra modifications (if some above checks ` if IsPlayerDead then` were required because of this only).

Reply:
@and3md convinced me -- we'll actually make `StateGameOver` transparent.